common:
    lid_topic:  "/livox/lidar"
    imu_topic:  "/livox/imu"
    radar_topic : "/radar_enhanced_pcl" 
    topic_twist : "/eagle_data/twist"
    topic_inlier_pc2 : "/eagle_data/inlier_pc2"
    topic_outlier_pc2 : "/eagle_data/outlier_pc2"

threshold:
    power_threshold: 0
    distance_near_thresh : 2
    distance_far_thresh : 100
    z_low_thresh : -1
    z_high_thresh : 100
    downsample_method : "VOXELGRID"   #VOXELGRID or APPROX_VOXELGRID or NONE
    downsample_resolution : 0.1
    outlier_removal_method : "RADIUS" #"STATISTICAL" or "RADIUS" or "NONE"
    statistical_mean_k : 30
    statistical_stddev : 1.2
    radius_radius : 0.5
    radius_min_neighbors : 1

cov :
    LASER_POINT_COV : 0.005
    VELOCITY_COV : 0.01 
    GRAVITY_COV : 0.005    

    
preprocess:
    lidar_type: 0           
    radar_type : 0
    n_scans: 6
    timestamp_unit: 3       # the unit of time/t field in the PointCloud2 rostopic: 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    feature_enabled: 0
    point_filter_num: 3
    blind: 2

mapping:
    # loss_threshold: 1000
    max_iteration: 4
    cube_side_length: 1000.0
    filter_size_surf: 0.5           # in some scenes with intense motion, reduce this value, such as set 0.15
    filter_size_map: 0.5           # in some scenes with intense motion, reduce this value, such as set 0.3
    acc_noise: 0.1
    gyr_noise: 0.1
    acc_bias_noise: 0.0001
    gyr_bias_noise: 0.0001
    fov_degree: 90
    det_range: 450.0
    extrinsic_est_en: false      # true: enable the online estimation of IMU-LiDAR extrinsic
    extrinsic_t_il: [0,0,0] 
    extrinsic_R_il: [1, 0, 0,
                    0, 1, 0,
                    0, 0, 1]
    extrinsic_T_ri : [   1, 0, 0, 0,
                            0, 1, 0, 0,
                            0, 0, 1, 0,
                            0, 0, 0, 1]
    extrinsic_T_rl : [   1, 0, 0, 0,
                        0, 1, 0, 0,
                        0, 0, 1, 0,
                        0, 0, 0, 1]

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
    traj_save_path: /home/vdcl/Downloads/PCD